recommenders/datasets/mock/movielens.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+Mock dataset schema to generate fake data for testing use. This will mimic the Movielens Dataset
```

Comment:
@laserprec can we discuss the pros and cons of having this function on a different file vs having it in the movielens.py file?

Reply:
i think we might be able to simplify a few things if we just push this directly into movielens.py

Reply:
I think this is my mental activity planning for this change 😄:

The sentimental part of me is like:
It pains me to add another 100+ lines to an already long 500 document.

The analytics part is like:
1. The new folder structure can contain other "mock" dataset schemas besides `movielens` (although one can argue that this over engineering for something that may not happen)
2. A clear separation between the code for testing purpose and the normal intended use

I am okay with keeping it separated or combined. But here is a summary:
- If we think this mocking strategy can be applied to other datasets for optimization, I think keeping it separated is good.
- If we think this is over-engineering and unnecessary complexity I think we can push into movielens.py

Reply:
@laserprec I think we should do the approach that is easier for users, what approach do you think they would prefer the most?

Reply:
I suspect the users of this code would be folks developing with the repo. Since this mocking strategy is only applied to movielens dataset, it would be better to keep it together and avoid adding more folder complexity without an immediate clear scenario for reusability. I can refactor the code and push into `movielens.py`.

Reply:
Done